1_fetch.R

```diff
@@ -88,12 +88,35 @@ p1_targets_list <- list(
     p1_reaches_sf,
     st_read(p1_reaches_shp)
   ),
+
+  # Download NLCD datasets 
+  tar_target(
+    p1_NLCD_data_zipped, 
+    download_NHD_NLCD_data(sb_id = sb_ids_NLCD,
+                           out_path = '1_fetch/out',
+                           downloaded_data_folder_name = NLCD_folders
+    ),
+    format = 'file'),
+  
+  ## Unzip all NLCD downloaded datasets 
+  ## Note - this returns a string or vector of strings of data path to unzipped datasets 
+  tar_target(p1_NLCD_data_unzipped, 
+             unzip_NHD_NLCD_data(downloaded_data_folder_path = p1_NLCD_data_zipped,
+                                 create_unzip_subfolder = T),
+             format = 'file'
+  ),
+  
+  ## read in NLCD datasets and subet by comid in DRB
+  ## Note that this returns a vector of dfs if more than one NLCD data is in the p1_NLCD_data_unzipped
+  tar_target(p1_NLCD_df,
+             read_subset_LC_data(LC_data_folder = p1_NLCD_data_unzipped, 
+                                 Comids_in_AOI_df = p1_nhdv2reaches_sf %>% select(COMID), 
```

Comment:
Note here that the p1_nhdv2reaches_sf target (a df) is pulled here to filter the COMID of our AOI.

Reply:
Additionally, p1_NLCD_df is a list of dfs if more than one sb_id is given, returns a simple df if only one sb_id is given.